azure-kusto-ingest/azure/kusto/ingest/descriptors.py

```diff
@@ -171,14 +173,20 @@ def upload_from_different_descriptor(
         :return new BlobDescriptor instance
         """
         blob_name = "{db}__{table}__{guid}__{file}".format(db=database, table=table, guid=descriptor.source_id, file=descriptor.stream_name)
-        random_container = random.choice(containers)
-        try:
-            blob_service = BlobServiceClient(random_container.account_uri, proxies=proxy_dict)
-            blob_client = blob_service.get_blob_client(container=random_container.object_name, blob=blob_name)
-            blob_client.upload_blob(data=stream, timeout=timeout)
-        except Exception as e:
-            raise KustoBlobError(e)
-        return BlobDescriptor(blob_client.url, descriptor.size, descriptor.source_id)
+
+        retries_left = min(max_retries, len(containers))
+        for container in containers:
+            try:
+                blob_service = BlobServiceClient(container.account_uri, proxies=proxy_dict)
+                blob_client = blob_service.get_blob_client(container=container.object_name, blob=blob_name)
+                blob_client.upload_blob(data=stream, timeout=timeout)
+                resource_manager.Report_resource_usage_result(container.storage_account_name, True)
+                return BlobDescriptor(blob_client.url, descriptor.size, descriptor.source_id)
+            except Exception as e:
+                retries_left = retries_left - 1
```

Comment:
Add a TODO to log the retries once we have logs

Reply:
What do you mean? adding to TODO inside Report_resource_usage_result?

Reply:
```suggestion
                retries_left = retries_left - 1
                # TODO: log the retry once we have a proper logging system
```

That's it